Problem.h

```diff
@@ -48,31 +63,72 @@ class Problem {
 	virtual std::vector<std::pair<std::string, State> > expand(const State& s) const =0;
 	const State& getStartState() { return startState; }
 
+
+
+	bool checkIfSeen(Node<State>* com, std::vector<Node<State>* > seen){
```

Comment:
inefficient to pass vector by value
should compare states rather than Node<State>